src/Components/Todo/TodoItem.tsx

```diff
@@ -0,0 +1,62 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  onTodoRemove: (todoId: number) => void;
+  isRemovingCompleted: boolean;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onTodoRemove,
+  isRemovingCompleted,
+}) => {
+  const [isRemoving, setIsRemoving] = useState<boolean>(false);
+  const isNewTodo = todo.id === 0;
+
+  const isRemovingCompletedTodo = isRemovingCompleted && todo.completed;
+  const isLoading = isNewTodo || isRemoving || isRemovingCompletedTodo;
+
+  const handleRemoveTodo = () => {
+    onTodoRemove(todo.id);
+    setIsRemoving(true);
+  };
+
+  return (
+    <div
+      className={classNames(
+        'todo',
+        { completed: todo.completed },
+      )}
+    >
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+        />
+      </label>
+
+      <span className="todo__title">{todo.title}</span>
+
+      <button
```

Comment:
Add a disabled attribute to the button element to prevent multiple clicks when removing a todo. Example: disabled={isLoading}